src/core/SampleBuffer.cpp

```diff
@@ -1521,6 +1542,7 @@ void SampleBuffer::loadFromBase64(const QString & data)
 
 	m_audioFile = QString();
 	update();
+	Engine::getSampleBufferCache()->put(data.toStdString(), this);
```

Comment:
The string `data` can be quite huge, especially if it contains audio data that's encoded as base64. Therefore it's not a good idea to use it as a key.

Reply:
What do you suggest should be done alternatively? I'm having trouble figuring out what key should be used for ``SampleBuffer``'s that originate from Base64 data. I'm also hoping that using ``std::string`` as a key for file paths are fine, since the data is not too large, but let me you know if you find that to be a problem as well.

Reply:
In the current code the base64 path seems to be hit in two cases:
* In `AudioEngine::handleMetronome` when a play handle for a metronome sample is added (of which there are only two different ones)
* In `FileBrowserTreeWidget::previewFileItem` when a sample is previewed

The question is if it is really a good idea to fill the sample cache in these cases. Imagine a user that preview lots of files but uses only very few of them (if any). In that case the not used preview samples would "pollute" the cache with lots of data that's not really needed in the actual song.